dmg/dmg.c

```diff
@@ -33,13 +32,56 @@ int buildInOut(const char* source, const char* dest, AbstractFile** in, Abstract
 	return TRUE;
 }
 
-int usage(const char *name) {
+void usage(const char *name) {
 	printf("usage: %s (OPTIONS) [extract|build|build2048|res|iso|dmg|attribute] <in> <out> (partition)\n", name);
 	printf("OPTIONS:\n");
 	printf("\t-k\tkey\n");
 	printf("\t-J\tcompressor name (%s)\n", compressionNames());
 	printf("\t-L\tcompression level\n");
-	return 2;
+	exit(2);
+}
+
+/* Mini getopt. Contrary to macOS's getopt, allow options after positionals */
```

Comment:
Do we actually need this? Or can we just adjust usage to always put options before positionals?

Reply:
I considered this, but though it was a bit awkward for arguments to need to come even before commands. IE, you can't do `dmg build -J gzip ...`, it would have to be `dmg -J gzip build ...`. I agree it feels a bit unnecessary to have our own getopt implementation.

I'll look into a couple of easier options:
1. Maybe `getopt_long()` on macOS behaves more like GNU's getopt?
2. Maybe users on macOS who want nicer arg-parsing can link against some other getopt-providing library?

Reply:
That's a fair point, yeah. `dmg -J gzip build` is...awkward. I'm with either of those options, or if it proves to be too unwieldy, the current code is fine too.